start_control_ssm.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+log() {
+  echo "$*" >&2
+}
+
+enable_hybrid_env_ssm() {
+  # SSM parameters necessary to register hybrid environment Linux
+  failed_ssm_params=0
+  activation_code=$(get_user_data_ssm "activation-code")
+  activation_id=$(get_user_data_ssm "activation-id")
+  region=$(get_user_data_ssm "region")
+
+  # If any of the parameters are missing, then exit 1.
+  if [[ "${failed_ssm_params}" -ne 0 ]]; then
+    log "Missing required parameters to register with AWS Systems Manager (SSM)"
+    exit 1
+  fi
+
+  # Register with AWS Systems Manager (SSM)
+  amazon-ssm-agent -register -code "${activation_code}" -id "${activation_id}" -region "${region}"
+  if [[ "$?" -eq 1 ]]; then
+    log "Failed to register with AWS Systems Manager (SSM)"
+    exit 1
+  fi
+}
+
+# Extract the values from user-data json and count the failures (if any)
+get_user_data_ssm() {
+  local param_value
+  local ssm_param="${1:?}"
+  if ! param_value=$(jq --arg ssm_param "${ssm_param}" -e -r '.["ssm"][$ssm_param]' "${USER_DATA}"); then
+    log "Failed to parse ${ssm_param} from ${USER_DATA}"
+    ((++failed_ssm_params))
+    return 1
+  fi
+  echo "${param_value}"
+}
+
+main() {
+  # If /.bottlerocket/host-containers/control/user-data exists and is not empty
+  # and the symlinked /var/lib/amazon/ssm/registration file is not populated,
+  # then check to see if user-data contains ssm at the top-level. If so, attempt
+  # to manually register with SSM as a hybrid environment host.
+  declare -r PERSISTENT_CONTROL_DIR="/.bottlerocket/host-containers/control"
+  declare -r SSM_AGENT_LOCAL_STATE_DIR="/var/lib/amazon/ssm"
+  declare -r SSM_AGENT_PERSISTENT_STATE_DIR="${PERSISTENT_CONTROL_DIR}/ssm"
```

Comment:
Super nit: Can we bump this variable up one line so it's directly under the variable it's using (PERSISTENT_CONTROL_DIR)?  IMHO, makes it a tiny bit easier to follow the variable chaining.